cmd/cloner/cloner.go

```diff
@@ -122,8 +122,7 @@ func gen(buf *bytes.Buffer, it *codegen.ImportTracker, typ *types.Named) {
 		case *types.Slice:
 			if codegen.ContainsPointers(ft.Elem()) {
 				n := it.QualifiedName(ft.Elem())
-				writef("if src.%s != nil {", fname)
-				writef("dst.%s = make([]%s, len(src.%s))", fname, n, fname)
+				writef("dst.%s = append([]%s(nil), make([]%s, len(src.%s))...)", fname, n, n, fname)
```

Comment:
doesn't this end up allocating two slices? does the compiler optimize it away?

cc @josharian

Reply:
See https://go-review.googlesource.com/c/go/+/109517 :)

And for lots more discussion, https://github.com/golang/go/issues/24204 and https://github.com/golang/go/issues/21266

Reply:
thanks!